dotenv_cli/core.py

```diff
@@ -84,4 +85,13 @@ def run_dotenv(filename, command):
         env=env,
     )
     _, _ = proc.communicate()
+
+    def terminate_child(target=proc):
+        target.terminate()
+        sleep(3)
+        if target.poll() is not None:
+            logger.warning('Process survived the SIGTERM, killing it!')
+            target.kill()
+        atexit.register(terminate_child)
```

Comment:
I think this is not properly indented. Also I believe the `register` call needs to come before the **blocking** `proc.communicate` call.

I fixed it slightly different, I `kill` immediately as the child processes receive the same signals as the parents. So if the parent is quitting and the child is not, then it clearly ignored the signals. So at this point we can `kill` straightaway.

Reply:
Ah yes of course, the atexit had to be at the same level of the function definition -_-